llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -592,9 +592,12 @@ SchedulingStage InterBlockScheduling::updateScheduling(BlockState &BS) {
 
   // The loop schedule has converged, so we could declare our work done.
   // But first try SWP
-  if (BS.getRegions().size() == 1 && BS.getPostSWP().canAccept(*BS.TheBlock)) {
-    BS.FixPoint.II = 1;
-    return BS.FixPoint.Stage = SchedulingStage::Pipelining;
+  if (BS.getRegions().size() == 1) {
+    auto &PostSWP = BS.getPostSWP();
+    if (PostSWP.canAccept(*BS.TheBlock)) {
+      BS.FixPoint.II = PostSWP.getResMII(*BS.TheBlock);
```

Comment:
This saves a lot of work. II divides the linear schedule length to determine the number of copies in the DAG, and DAG construction is expensive and worse than linear in the number of nodes. Something like Conv2D immediately hits the target II, so we would also save 7 iterations of the main loop.